internal/stats/metrics_recorder_list.go

```diff
@@ -49,50 +49,46 @@ func NewMetricsRecorderList(shs []stats.Handler) *MetricsRecorderList {
 	}
 }
 
-func (l *MetricsRecorderList) RecordInt64Count(handle *estats.Int64CountHandle, incr int64, labels ...string) {
-	if got, want := len(handle.Labels)+len(handle.OptionalLabels), len(labels); got != want {
-		logger.Infof("length of labels passed to RecordInt64Count incorrect got: %v, want: %v", got, want)
+func verifyLabels(labels []string, optionalLabels []string, labelsRecv ...string) {
+	if got, want := len(labelsRecv), len(labels)+len(optionalLabels); got != want {
+		logger.Fatalf("Length of labels passed to Record incorrect, got: %v, want: %v.", got, want)
```

Comment:
IMO delete the logger now and call `panic(fmt.Sprintf)` here.

Also if we pass the descriptor let's print the metric name with `%q` as well.  Maybe lose the `got, want` formatting which I believe is unique to Go test assertions?

Reply:
Got rid of got, want. Recorded the metric name with %q from desc.

Reply:
Oh I didn't mean to remove the counts..  Just to turn it into a more english-y thing than a Go test error message.  E.g.

Like `Received %d labels in call to record metric %q, but expected %d`

Reply:
Ah ok noted switched to that.